### PR TITLE
Fix no timeout for network loading status in network store

### PR DESCRIPTION
**Pull request by oogetyboogety**
This pr creates a ErrorLoadingScreen ui component alongside the current LoadingScreen component.

things to do:
- [x] finish troubleshooting why `setRpcTarget` no longer hides the overlay
   This was an issue with the `currentView` not being updated and it being set to `accounts` instead of `unlock` or `config`. Instead, I've changed the ErrorLoadingScreen component to only render based on the route in `this.props.location.pathname` instead of the `currentView` prop.
- [x] flesh out the transaction flow through a FSM of the new redux props and `EventEmitter` states
   As an example, I've added the requirement that the newly specified overlay will only display while unlocked whereas the loading overlay does display regardless. Otherwise, the user may expect something to happen when they try to click on the `Set Custom RPC` option from the network dropdown on the `Switch network` button, however they will be redirected to the `Settings` page *behind* the overlay and only after they unlock the account. Let me know if the expected behavior should be different.
     Optional subtasks of this:
- [ ] -> add the showErrorLoadingScreen action to the API hook for changing the provider config
- [ ] -> forward user to accounts page from settings page after setting a custom RPC target
- [ ] -> Move `validateRpc` from a function internal to the Settings page component to a dispatch-able action of the network component. I think this should probably be done, but maybe moved into a new issue.
- [x] add the showErrorLoadingScreen action after setting a custom RPC target similar to the built in provider configs in the network dropdown menu 
- [x] Add close button to new overlay
- [x] internationalize text on buttons
- [x] style buttons 
- [ ] change the button class names from the `hero balance` button classnames
- [ ] cleanup logging and lint

https://github.com/MetaMask/metamask-extension/issues/3776

### CODE REVIEW

**Comment by whymarrh** [comment]
@oogetyboogety thanks for the PR! Would you mind updating adding a description of the changes to the PR and maybe include what's still a work-in-progess?

**Comment by oogetyboogety** [comment]
@whymarrh No problem, i updated the description and I am continuing to work on this

**Comment by oogetyboogety** [comment]
@whymarrh WIP finished, I think it's ready for review

**Comment by bdresser** [comment]
Looks good @oogetyboogety! Thanks for your work.

Couple small things
* Could you increase the opacity of the modal? Should be somewhere around 90-95%
* Can you make the "Oops! Something..." text black and decrease the font size to 18px
* `X` in the top-right should be larger and black as well - gets lost a bit
* @cjeria do you think we need more padding on the "Switch Networks" button? 
* I noticed that on my first attempt the timeout was quite short (~5 seconds) but on later attempts it was much longer (~25 seconds). Ideally they're all consistent around ~15 seconds

screen shot 2018-09-10 at 12 23 57 pm

**Comment by cjeria** [comment]
@oogetyboogety I agree with bobby, no need to increase padding of the network pill. Since it's all sort of blending in together making hard to read, how about we put the full message + buttons in a dialog box for better readability. See screenshot

switching network - timeout message

**Comment by bdresser** [comment]
thanks @oogetyboogety! could you post a screenshot for @cjeria please?

@whymarrh or @danjm, could you take a look at the implementation?

File: app/scripts/controllers/network/network.js [hunk 1]
```diff
@@ -39,19 +39,23 @@ module.exports = class NetworkController extends EventEmitter {
...
+    this.store = new ComposedStore({ provider: this.providerStore, network: this.networkStore, change: this.changeStore })
     this.on('networkDidChange', this.lookupNetwork)
+    this.on('networkTimeout', this.timeoutNetwork)
     // provider and block tracker
     this._provider = null
     this._blockTracker = null
     // provider and block tracker proxies - because the network changes
     this._providerProxy = null
     this._blockTrackerProxy = null
+      console.log('constructing')
```
Comment (bdresser) [inline app/scripts/controllers/network/network.js]: remove

File: app/scripts/controllers/network/network.js [hunk 2]
```diff
@@ -39,19 +39,23 @@ module.exports = class NetworkController extends EventEmitter {
...
   }
 
   initializeProvider (providerParams) {
     this._baseProviderParams = providerParams
     const { type, rpcTarget } = this.providerStore.getState()
+    log.warn('intializing')
```
Comment (bdresser) [inline app/scripts/controllers/network/network.js]: remove?

**Comment by bdresser** [comment]
@oogetyboogety i just went to test this out and am seeing some weird behavior - the extension will connect to a "private network" no matter what Custom RPC I enter.

Also, there are a lot of console.logs left around the code.

Could you take another look and then please post a screenshot of the updated UI for our designer? Thanks!

**Comment by oogetyboogety** [comment]
@bdresser Could you copy and paste your Metamask UI state log? I would like to see if the events are firing for both the Error Loading Network timeout modal action and the show modal action and if there are any exceptions in the log.

Thanks!

**Comment by oogetyboogety** [comment]
@bdresser These two would probably be most helpful:
UI State Logs (Chrome)

If you've had an issue sending transactions or found a visual glitch in the rendering of MetaMask, we might ask you for a UI State log. To get one of these:

    Open MetaMask and ensure its unlocked.
    Click the menu in the top right, click "Settings".
    Click the button that says "Copy State Logs".
    It will copy something that looks like this.
    Upload it as a gist or to some other pastebin site
    Send the link to us, ideally in a new Github issue.

Popup Logs (Chrome)

To get logs from the popup:

    Click the MetaMask fox in the top right of your browser.
    Wait for the popup to open (or partially open if that's part of the bug).
    Right-click in the newly opened popup, and select Inspect Element.
    Wait for the new Inspector window to open.
    Click Console at the top of the Inspector window.
    Look for any strange logs, especially red errors!

Sorry for the extra work.

**Comment by bdresser** [comment]
Thanks for the links 😉 

State logs are [here](https://gist.github.com/bdresser/41abe58d2449a92da68000df75c5fe9a)

See this in my background console:

screen shot 2018-10-02 at 9 49 35 am

This did actually work on the third try of connecting to a non-existend Custom RPC - I finally got the popup. But when I clicked "try again" it just went ahead an connected to a Private Network (even though there was none running).

I think what happens is: if I select a network from the drop-down, it will have the expected time-out behavior. If I add a *new* network, it will spin and eventually connect, even if the network doesn't exist.

Also the UI still needs some work - the buttons are too small and not styled properly, typeface is too small, and you picked a different emoji? 

Also, as described in the original spec, we should no longer show the standalone spinner in the top-right. It should spin inside the network pill next to the string "Private Network"

screen shot 2018-10-02 at 9 49 20 am

**Comment by bdresser** [comment]
@oogetyboogety what's up? you still working on this?

**Comment by oogetyboogety** [comment]
@bdresser I'll take another look over the weekend,i think the user flow is not fully specified in the issue. It works for the flow in the issue for me and the ui isn;t all screwed up like yours.

**Comment by oogetyboogety** [comment]
Also whats the character code in the emoji

**Comment by oogetyboogety** [comment]
@bdresser gotcha it should not join the network was what i was missing. But it's weird that it doesn't show up at all for you the first couple times...I was sure i added my event to show the timeout modal after 30 seconds of loading to the settings button, but let me check. I think I will rework how the modal appears anyway to be like a standard modal. I will take a look at the network name in the pill as well...that may require storing the network name separately as originally planned above before I started. I'll look into that as well. I'll keep you posted!

**Comment by whymarrh** [comment]
@oogetyboogety how do you feel about this PR? Is there anything we can do on our end to help out? No rush, just want to make sure this doesn't fall too far behind `develop` and end with with too many conflicts.

**Pull request closed without merging**